addons/mail/static/src/views/form/form_compiler.js

```diff
@@ -0,0 +1,125 @@
+            case 'message_follower_ids':
+                hasFollowers = true;
+                hasParentReloadOnFollowersUpdate = Boolean(options['post_refresh']);
+                isAttachmentBoxVisibleInitially = isAttachmentBoxVisibleInitially || Boolean(options['open_attachments']);
+                break;
+            case 'message_ids':
+                hasMessageList = true;
+                hasParentReloadOnAttachmentsChanged = options['post_refresh'] === 'always';
+                hasParentReloadOnMessagePosted = Boolean(options['post_refresh']);
+                isAttachmentBoxVisibleInitially = isAttachmentBoxVisibleInitially || Boolean(options['open_attachments']);
+                break;
+        }
+    }
+    const chatterContainerXml = createElement("ChatterContainer");
+    chatterContainerXml.setAttribute("hasActivities", hasActivities);
+    chatterContainerXml.setAttribute("hasFollowers", hasFollowers);
+    chatterContainerXml.setAttribute("hasMessageList", hasMessageList);
+    chatterContainerXml.setAttribute("hasParentReloadOnAttachmentsChanged", hasParentReloadOnAttachmentsChanged);
+    chatterContainerXml.setAttribute("hasParentReloadOnFollowersUpdate", hasParentReloadOnFollowersUpdate);
+    chatterContainerXml.setAttribute("hasParentReloadOnMessagePosted", hasParentReloadOnMessagePosted);
+    chatterContainerXml.setAttribute("isAttachmentBoxVisibleInitially", isAttachmentBoxVisibleInitially);
+    chatterContainerXml.setAttribute("threadId", "props.record.resId or undefined");
+    chatterContainerXml.setAttribute("threadModel", "props.record.resModel");
+    chatterContainerXml.setAttribute("webRecord", "props.record");
+
+    const chatterContainerHookXml = createElement("div");
+    chatterContainerHookXml.classList.add("o_FormRenderer_chatterContainer");
+    append(chatterContainerHookXml, chatterContainerXml);
+    return chatterContainerHookXml;
+}
+
+function compileAttachmentPreview(node, params) {
+    const webClientViewAttachmentViewContainerHookXml = createElement("div");
+    webClientViewAttachmentViewContainerHookXml.classList.add('o_attachment_preview');
+    const webClientViewAttachmentViewContainerXml = createElement("WebClientViewAttachmentViewContainer");
+    webClientViewAttachmentViewContainerXml.setAttribute("threadId", "props.record.resId or undefined");
+    webClientViewAttachmentViewContainerXml.setAttribute("threadModel", "props.record.resModel");
+    append(webClientViewAttachmentViewContainerHookXml, webClientViewAttachmentViewContainerXml);
+    return webClientViewAttachmentViewContainerHookXml;
+}
+
+registry.category("form_compilers").add("chatter_compiler", {
+    selector: "div.oe_chatter",
+    fn: compileChatter,
+});
+
+registry.category("form_compilers").add("attachment_preview_compiler", {
+    selector: "div.o_attachment_preview",
+    fn: compileAttachmentPreview,
+});
+
+patch(FormCompiler.prototype, 'mail', {
+    compile() {
+        // TODO no chatter if in dialog?
```

Comment:
this kind of things should be done either with a t-if on the node, or with a t-if inside the component's template itself. Since a component can have 0 root node as of owl2, this is feasible.

Reply:
The main question is how to know when the form is in a dialog?